vm/src/stdlib/collections.rs

```diff
@@ -472,6 +472,32 @@ mod _collections {
             self.borrow_deque().len()
         }
 
+        #[pymethod(magic)]
+        fn add(&self, other: PyObjectRef, vm: &VirtualMachine) -> PyResult<Self> {
+            if other.payload_is::<PyDeque>() {
+                let mut deque = self.borrow_deque().clone();
```

Comment:
sorry for not anchoring. we can make a new deque instance and use self.extend function to it

Reply:
and to prevent the fmt&clippy ci, before commit the code run `cargo fmt --all` and `cargo clippy`

Reply:
Consider the following though @eldpswp99, in `extend` we have to deal with any object as long as we can grab an iterator from it to extract its elements. Here we only deal with `deque`s, as such, we can directly clone the underlying `VecDeque` from `other` and get a significant speed-up (still O(N) but with significantly smaller lower-order terms).